### ISSUE TITLE
[Main script] - main.py producing error for some arguments

### ISSUE THREAD

**Issue by EvangelosTikas** [ludocomito/antibody_training_pipeline_ESM#6]
In an attempt to run the `main.py` I noticed that the config needs the model_save_path specified otherwise it returns an error. If this is intentional we can skip this bug.

```

!python main.py --setup-dirs
Using configuration file: config.yaml
Configuration validated successfully: config.yaml
============================================================
ANTIBODY CLASSIFICATION PIPELINE
============================================================
ESM Model: facebook/esm1v_t33_650M_UR90S_1
Device: mps
Classifier: logistic_regression
Random State: 42
Max Iterations: 1000
Cross-validation Folds: 10
Metrics: accuracy, precision, recall, f1, roc_auc
Log Level: INFO
Training Data: ./train_datasets/VH_only_training_ready.csv
============================================================

Error: 'model_save_path'

For help, run: python main.py --help
```

**Plus** when running the `main.py` the  `--multi-seed` does not exists yet in this file as an argument, therefore 
resulting in an error:

Setup:
Google Colab

GPU: T4 GPU
GPU RAM : 15.0 GB
Python: Python 3.12.11
System RAM : 12.7 GB

packages installed:

```
Using Python 3.12.11 environment at: /content/.venv
Resolved 59 packages in 233ms
Installed 59 packages in 481ms
 + certifi==2025.10.5
 + charset-normalizer==3.4.3
 + contourpy==1.3.3
 + cycler==0.12.1
 + filelock==3.19.1
 + fonttools==4.60.1
 + fsspec==2025.9.0
 + hf-xet==1.1.10
 + huggingface-hub==0.35.3
 + idna==3.10
 + jinja2==3.1.6
 + joblib==1.5.2
 + kiwisolver==1.4.9
 + markupsafe==3.0.3
 + matplotlib==3.10.6
 + mpmath==1.3.0
 + narwhals==2.7.0
 + networkx==3.5
 + numpy==2.3.3
 + nvidia-cublas-cu12==12.8.4.1
 + nvidia-cuda-cupti-cu12==12.8.90
 + nvidia-cuda-nvrtc-cu12==12.8.93
 + nvidia-cuda-runtime-cu12==12.8.90
 + nvidia-cudnn-cu12==9.10.2.21
 + nvidia-cufft-cu12==11.3.3.83
 + nvidia-cufile-cu12==1.13.1.3
 + nvidia-curand-cu12==10.3.9.90
 + nvidia-cusolver-cu12==11.7.3.90
 + nvidia-cusparse-cu12==12.5.8.93
 + nvidia-cusparselt-cu12==0.7.1
 + nvidia-nccl-cu12==2.27.3
 + nvidia-nvjitlink-cu12==12.8.93
 + nvidia-nvtx-cu12==12.8.90
 + packaging==25.0
 + pandas==2.3.3
 + pillow==11.3.0
 + plotly==6.3.1
 + pyparsing==3.2.5
 + python-dateutil==2.9.0.post0
 + pytz==2025.2
 + pyyaml==6.0.3
 + regex==2025.9.18
 + requests==2.32.5
 + safetensors==0.6.2
 + scikit-learn==1.7.2
 + scipy==1.16.2
 + seaborn==0.13.2
 + setuptools==80.9.0
 + six==1.17.0
 + sympy==1.14.0
 + threadpoolctl==3.6.0
 + tokenizers==0.22.1
 + torch==2.8.0
 + tqdm==4.67.1
 + transformers==4.57.0
 + triton==3.4.0
 + typing-extensions==4.15.0
 + tzdata==2025.2
 + urllib3==2.5.0
/usr/bin/python3
```

[Antibody.ipynb](https://github.com/user-attachments/files/22749245/Antibody.ipynb)

**Comment by ludocomito** [comment]
Thanks for the thorough description.
- The `model_save_path` argument is required as the code automatically stores the model checkpoint at the end of each run.
- I removed the `--multi_seed` mention from USAGE.md, as this was part of an old version of the code.

**Comment by EvangelosTikas** [comment]
Since the `model_save_path` is not mentioned in the `USAGE.md` or `README.md` there could be something explaining that the user must explicitly give that in his configuration yaml since this does not exist in your example `config.yaml`. This is just to help other people replicate this.
